### ISSUE TITLE
Add timeline ruler into main keyboard navigation hierarchy

### ISSUE THREAD

**Issue by DilsonsPickles** [audacity/audacity#11367]
### Description

The timeline ruler needs to be added into the main keyboard navigation hierarchy following the "Add track" button and preceding the first track. 

Pressing shift + f10 with the timeline ruler in focus will summon the right-click context menu. 

With keyboard focus on the timeline ruler, the User should be able to use the left/right arrow key shortcuts to move the playhead left and right. All modifiers should also be able to applied when performing these left/right arrow key moves.

<img width="1512" height="949" alt="Image" src="https://github.com/user-attachments/assets/a934bb87-c218-474a-896a-dbfb1434b202" />

**Comment by dozzzzer** [comment]
Tested
